### PR TITLE
UnicastZenPing shouldn't ping the address of the local node

### PR DESCRIPTION
**Pull request by bleskes**
Pinging the local node address doesn't really add to discovering other nodes. It just pollutes the logs with unneeded information.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM.

File: core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java [hunk 1]
```diff
@@ -214,6 +215,9 @@ public UnicastZenPing(Settings settings, ThreadPool threadPool, TransportService
         final List<Future<TransportAddress[]>> futures =
             executorService.invokeAll(callables, resolveTimeout.nanos(), TimeUnit.NANOSECONDS);
         final List<DiscoveryNode> discoveryNodes = new ArrayList<>();
+        final Set<TransportAddress> localAddresses = new HashSet<>();
+        localAddresses.add(transportService.boundAddress().publishAddress());
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java]: Your CI run is getting an NPE here.

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java]: I think it's similar to how you had to override `TransportSerivce#boundAddress` in the tests.

**Comment by bleskes** [comment]
Thx @jasontedor

**Pull request merged**